ayon_bin_distro/lakectlpy/wrapper.py

```diff
@@ -122,7 +132,43 @@ def help(self):
                 continue
             sys.stdout.write(process.stdout.readline())
 
-    def list_repo_objects(self, lake_fs_repo_uri: str)-> List[str]:
+    def list_repos(self) -> List[dict]:
+        """lists out all the repos the user can see on a given LakeFs server
+
+        Raises:
+            DialTcpError: raised if an connection cant be established.
+
+        Returns: a list of repos in a dict representation.
+
+        """
+        process = self._run(["repo", "list"], non_blocking_stdout=False)
+        repos: List[dict] = []
+
+        while process.poll() is None:
+            reads = [process.stdout, process.stderr]
+            readable, _, _ = select.select(reads, [], [])
+
+            if process.stderr in readable:
+                stderr_output = process.stderr.readline()
+                if stderr_output:
+                    raise DialTcpError(stderr_output)
```

Comment:
> this can be a 401 with an message that your not logged in.
> or an tcp dial error if the uri cant be found.

So technically this may NOT be a `DialTcpError`? If so, should we generalize it to `LakeFSConnectionError` or whatever?

Reply:
yes i will add as an todo.